generators/server/files.js

```diff
@@ -374,6 +374,10 @@ const serverFiles = {
                 {
                     file: 'package/security/jwt/JWTConfigurer.java',
                     renameTo: generator => `${generator.javaDir}security/jwt/JWTConfigurer.java`
+                },
+                {
+                    file: 'package/config/apidoc/SwaggerSecurityCustomizer.java',
+                    renameTo: generator => `${generator.javaDir}config/apidoc/SwaggerSecurityCustomizer.java`
```

Comment:
@Hawkurane thanks for the review. The reason why I didn't modify the jhipster-frameworks is because the security configuration depends on what the user chooses during generation. 

- a monolith has some private endpoints and some public endpoints
- a microservice has only private endpoints by default (all protected)
- the security definitions in the yml file when using jwt (Bearer) are different from the security definition we should define when using oauth or UAA

For what I know jhipster-framework should not be affected by what the user chooses during generation, and the only way to "dynamically" configure beans in the jhipster frameworks is to use spring conditional annotations. If we find how to use those annotations, we could think about moving SpringFox SecurityConfiguration, SecuritySchemes, and SecurityContext in the jhipster-framework, but we would miss the public/private endpoint part. 

Let me know what you think

Reply:
I think these are very valid points to consider. What about customizing the docket in 2 layers aka:

- First customization via jhipster/jhipster 
- Second one in a generated file (with SecurityConfiguration, SecuritySchemes and SecurityContext) for swagger ui v3, if we can keep this file I suggest we change the name for better clairty (e.g. SwaggerSecurityCustomizer, ...)

Another approach would migrate the swagger customizer completely from the jhipster frameworks to the generation as separating the two classes that essentially do the same thing might be confusing for the dev experience.

I am unsure as what could be considered as best practice so I am looking out for your response

Reply:
I'd keep it like it is now and see if some suggestions come out from the reviewers.